streams/src/main/java/org/apache/kafka/streams/Topology.java

```diff
@@ -656,6 +660,14 @@ public synchronized Topology addProcessor(final String name,
                                               final ProcessorSupplier supplier,
                                               final String... parentNames) {
         internalTopologyBuilder.addProcessor(name, supplier, parentNames);
+        final Set<StoreBuilder<?>> stores = supplier.stores();
+        if (stores != null) {
+            for (final StoreBuilder storeBuilder : stores) {
+                internalTopologyBuilder.addStateStore(storeBuilder, name);
+            }
+            final String[] storeNames = stores.stream().map(StoreBuilder::name).toArray(String[]::new);
+            internalTopologyBuilder.connectProcessorAndStateStores(name, storeNames);
```

Comment:
Why do we need this? Calling `addStateStore(storeBuilder, name)` should connect the store to the processor already?

Reply:
Good point, removing.